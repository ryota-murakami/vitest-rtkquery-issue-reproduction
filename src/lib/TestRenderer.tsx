import type { RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import type { ReactElement } from 'react'
import React from 'react'
import { Provider } from 'react-redux'
import type { MemoryRouterProps } from 'react-router'
import { MemoryRouter } from 'react-router-dom'

import { store } from './../redux/store'

const TestRenderer = (
  ui: ReactElement,
  memoryRouterProps?: MemoryRouterProps
): RenderResult => {
  const renderResult = render(
    <Provider store={store}>
      <MemoryRouter {...memoryRouterProps}>{ui}</MemoryRouter>
    </Provider>
  )
  return renderResult
}

export default TestRenderer
