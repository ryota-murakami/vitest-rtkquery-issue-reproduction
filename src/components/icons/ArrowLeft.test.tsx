import React from 'react'

import TestRenderer from '../../lib/TestRenderer'

import ArrowLeft from './ArrowLeft'

test('should render ArrowLeft', () => {
  const {
    container: { firstChild },
  } = TestRenderer(<ArrowLeft />)
  expect(firstChild).toBeTruthy()
  expect(firstChild).toMatchSnapshot()
})
