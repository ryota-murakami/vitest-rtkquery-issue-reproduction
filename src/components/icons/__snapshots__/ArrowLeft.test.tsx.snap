// Vitest Snapshot v1

exports[`should render ArrowLeft 1`] = `
<svg
  class="h-6 w-6"
  data-testid="arrow-left"
  fill="none"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M7 16l-4-4m0 0l4-4m-4 4h18"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
  />
</svg>
`;
